pkg/storage/backend-storage/backend-storage.go

```diff
@@ -107,6 +108,141 @@ func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachin
+	if migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == migration.Status.MigrationState.SourcePersistentStatePVCName {
+		// The migration either didn't actually start, or the backend storage is RWX.
+		// In both cases we consider the migration as failed.
+		migration.Status.Phase = corev1.MigrationFailed
+		return nil
+	}
+
+	// An interrupted migration exists. Using a job to check if the source PVC contains /meta/migrated,
+	// which would indicate that the libvirt migration finished.
+	// A JobComplete condition indicates the file is present, the migration was successful and the target PVC prevails
+	// A JobFailed condition indicated the file is absent, the migration didn't finish and the source PVC prevails
+
+	jobName := "recover-" + migration.Name
+
+	job, err := client.BatchV1().Jobs(vmi.Namespace).Get(context.Background(), jobName, metav1.GetOptions{})
+	if err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return err
+		}
+		job = buildRecoveryJob(jobName, migration.Status.MigrationState.SourcePersistentStatePVCName, launcherImage)
+		job, err = client.BatchV1().Jobs(vmi.Namespace).Create(context.Background(), job, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+		// The job was just created, return an error to be re-enqueued to check on the job
+		return fmt.Errorf("a migration recovery had to be initiated")
+	}
+
+	job, err = client.BatchV1().Jobs(job.Namespace).Get(context.Background(), job.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	for _, c := range job.Status.Conditions {
+		switch c.Type {
+		case batchv1.JobComplete:
+			if c.Status == v1.ConditionTrue {
+				err = MigrationHandoff(client, pvcStore, migration)
+				if err == nil {
+					migration.Status.Phase = corev1.MigrationSucceeded
+					_ = client.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{
+						PropagationPolicy: pointer.P(metav1.DeletePropagationBackground),
+					})
+				}
+				return err
+			}
+		case batchv1.JobFailed:
+			if c.Status == v1.ConditionTrue {
+				if c.Reason == batchv1.JobReasonPodFailurePolicy {
+					// The job ran properly but didn't find /meta/migrated, meaning the migration failed
+					err = MigrationAbort(client, migration)
+					if err == nil {
+						migration.Status.Phase = corev1.MigrationFailed
+						_ = client.BatchV1().Jobs(job.Namespace).Delete(context.Background(), job.Name, metav1.DeleteOptions{
+							PropagationPolicy: pointer.P(metav1.DeletePropagationBackground),
+						})
+					}
+					return err
+				} else {
+					return fmt.Errorf(c.Message)
+				}
+			}
+		default:
+			break
+		}
+	}
+
+	return fmt.Errorf("migration recovery job still running")
+}
+
+func buildRecoveryJob(jobName, pvcName, launcherImage string) *batchv1.Job {
```

Comment:
One thought, setting owner reference for the migration would allow us to not need gc the jobs.

Reply:
Good idea! Will do